Can't compile SU2 with GCC 4.8.5

Hi all! I am facing this issue when trying to do the ninja build.
I am trying to install SU2 on the remote server where I don't have root access.

**Firstly, I cloned the latest repo (per Nov 3, 2021), and then run the following meson command.**
`./meson.py build -Denable-autodiff=true -Denable-directdiff=true -Denable-pywrapper=true --prefix=/home/FDa/FDa045/programs/SU2/SU2-7.2.1 -Dwith-mpi=enabled -Denable-mkl=true -Dmkl_root=/home/FDa/FDa045/programs/intel/2020_update4/mkl`

**meson results:**

> Project name: SU2
Project version: 7.2.1 "Blackbird"
C compiler for the host machine: cc (gcc 4.8.5 "cc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-28)")
C linker for the host machine: cc ld.bfd 2.27-27
C++ compiler for the host machine: c++ (gcc 4.8.5 "c++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-28)")
C++ linker for the host machine: c++ ld.bfd 2.27-27
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/home/FDa/FDa045/anaconda3/bin/python3)
Found pkg-config: /usr/bin/pkg-config (0.27.1)
mpicc found: YES (/work/A/openmpi/3.0.0_intel2018u2/bin/mpicc) 3.0.0
Run-time dependency MPI for c found: YES 3.0.0
mpic++ found: YES (/work/A/openmpi/3.0.0_intel2018u2/bin/mpic++) 3.0.0
Run-time dependency MPI for cpp found: YES 3.0.0
Message: Boost sources found.
Found CMake: /home/FDa/FDa045/programs/cmake/cmake-3.17.3/bin/cmake (3.17.3)
Run-time dependency mkl-static-lp64-seq found: NO (tried pkgconfig and cmake)
Program swig found: YES (/usr/bin/swig)
Message: Using mpi4py from /home/FDa/FDa045/anaconda3/lib/python3.7/site-packages/mpi4py/include
Dependency python found: YES (pkgconfig)
Program install.sh found: YES (/bin/sh /home/FDa/FDa045/programs/SU2/SU2_PY/pySU2/install.sh)

**Next, I copied the 4-line output needed for .bashrc, and run ninja.**
`./ninja -j36 -C build install`

**Then, I got the following errors:**

> ninja: Entering directory `build'
[691/994] Compiling C++ object SU2_CFD/src/libSU2core_AD.a.p/solvers_CTurbSolver.cpp.o
FAILED: SU2_CFD/src/libSU2core_AD.a.p/solvers_CTurbSolver.cpp.o
c++ -ISU2_CFD/src/libSU2core_AD.a.p -ISU2_CFD/src -I../SU2_CFD/src -I../externals/CLI11 -I../externals/cgns/adf -Iexternals/cgns -I../externals/cgns -I../externals/metis/include -I../externals/metis/GKlib -I../externals/metis/libmetis -I../externals/parmetis/include -I../externals/parmetis/libparmetis -Iexternals/tecio -I../externals/tecio -Iexternals/tecio/teciompisrc -I../externals/tecio/teciompisrc -I/home/FDa/FDa045/programs/intel/2020_update4/mkl/include -I../externals/codi/include -I../externals/medi/include -I../externals/medi/src -ICommon/src -I../Common/src -I/work/A/openmpi/3.0.0_intel2018u2/include -pipe -D_FILE_OFFSET_BITS=64 -std=c++11 -O3 -fPIC -m64 -pthread -Wno-empty-body -Wno-unused-parameter -Wno-deprecated-declarations -Wno-non-virtual-dtor -DHAVE_CGNS -DNDEBUG -DHAVE_MPI -DHAVE_METIS -DHAVE_PARMETIS -DHAVE_TECIO -DTECIOMPI -DHAVE_MKL -DCODI_REVERSE_TYPE -MD -MQ SU2_CFD/src/libSU2core_AD.a.p/solvers_CTurbSolver.cpp.o -MF 'SU2_CFD/src/libSU2core_AD.a.p/solvers_CTurbSolver.cpp.o.d' -o SU2_CFD/src/libSU2core_AD.a.p/solvers_CTurbSolver.cpp.o -c ../SU2_CFD/src/solvers/CTurbSolver.cpp
In file included from ../SU2_CFD/src/solvers/CTurbSolver.cpp:31:0:
../SU2_CFD/src/solvers/../../include/solvers/CScalarSolver.inl: In instantiation of ‘CScalarSolver<VariableType>::CScalarSolver(bool) [with VariableType = CTurbVariable]’:
../SU2_CFD/src/solvers/CTurbSolver.cpp:34:16:   required from here
../SU2_CFD/src/solvers/../../include/solvers/CScalarSolver.inl:33:101: error: invalid initializer for array member ‘su2double CScalarSolver<CTurbVariable>::lowerlimit [2]’
 CScalarSolver<VariableType>::CScalarSolver(bool conservative) : CSolver(), Conservative(conservative) {}
                                                                                                     ^
../SU2_CFD/src/solvers/../../include/solvers/CScalarSolver.inl:33:101: error: invalid initializer for array member ‘su2double CScalarSolver<CTurbVariable>::upperlimit [2]’
../SU2_CFD/src/solvers/../../include/solvers/CScalarSolver.inl:33:101: error: invalid initializer for array member ‘su2double CScalarSolver<CTurbVariable>::Solution_Inf [2]’
../SU2_CFD/src/solvers/../../include/solvers/CScalarSolver.inl: In instantiation of ‘CScalarSolver<VariableType>::CScalarSolver(CGeometry*, CConfig*, bool) [with VariableType = CTurbVariable]’:
../SU2_CFD/src/solvers/CTurbSolver.cpp:34:16:   required from here
../SU2_CFD/src/solvers/../../include/solvers/CScalarSolver.inl:37:43: error: invalid initializer for array member ‘su2double CScalarSolver<CTurbVariable>::lowerlimit [2]’
     : CSolver(), Conservative(conservative) {
                                           ^
../SU2_CFD/src/solvers/../../include/solvers/CScalarSolver.inl:37:43: error: invalid initializer for array member ‘su2double CScalarSolver<CTurbVariable>::upperlimit [2]’
../SU2_CFD/src/solvers/../../include/solvers/CScalarSolver.inl:37:43: error: invalid initializer for array member ‘su2double CScalarSolver<CTurbVariable>::Solution_Inf [2]’
[726/994] Compiling C++ object SU2_CFD/src/libSU2core_AD.a.p/solvers_CEulerSolver.cpp.o
ninja: build stopped: subcommand failed.

**I suspect it got something to do with my GCC version. Since I can't update the system GCC, I tried to build the newer version of GCC locally and used it to build SU2. But then I encountered another systemic error that seem to be much more complicated (long errors, but one of them is..)**

> /home/FDa/FDa045/.local/lib64/libstdc++.so.6: no version information available (required by /usr/bin/swig) 

I am happy if anybody could suggest me anything!!

**The remote server systems**
 - OS: Linux (Red Hat Enterprise Linux Server 7.5 (Maipo))
 - C++ compiler and version: [g++ (GCC) 4.8.5]
 - MPI implementation and version: [OpenMPI 3.0.0]
 - SU2 Version: [e.g., v7.2.1]

--- a/SU2_CFD/include/solvers/CScalarSolver.hpp
+++ b/SU2_CFD/include/solvers/CScalarSolver.hpp
@@ -49,11 +49,11 @@ class CScalarSolver : public CSolver {
 
   unsigned long omp_chunk_size; /*!< \brief Chunk size used in light point loops. */
 
-  su2double lowerlimit[MAXNVAR] = {0.0}; /*!< \brief contains lower limits for turbulence variables. Note that ::min()
-                                            returns the smallest positive value for floats. */
-  su2double upperlimit[MAXNVAR] = {0.0}; /*!< \brief contains upper limits for turbulence variables. */
+  su2double lowerlimit[MAXNVAR]; /*!< \brief contains lower limits for turbulence variables. Note that ::min()
+                                             returns the smallest positive value for floats. */
+  su2double upperlimit[MAXNVAR]; /*!< \brief contains upper limits for turbulence variables. */
 
-  su2double Solution_Inf[MAXNVAR] = {0.0}; /*!< \brief Far-field solution. */
+  su2double Solution_Inf[MAXNVAR]; /*!< \brief Far-field solution. */
 
   const bool Conservative; /*!< \brief Transported Variable is conservative. Solution has to be multiplied with rho. */
 
@@ -171,11 +171,6 @@ class CScalarSolver : public CSolver {
   virtual void ComputeUnderRelaxationFactor(const CConfig* config) {}
 
  public:
-  /*!
-   * \brief Constructor of the class.
-   */
-  CScalarSolver(bool conservative);
-
   /*!
    * \brief Destructor of the class.
    */

--- a/SU2_CFD/include/solvers/CScalarSolver.inl
+++ b/SU2_CFD/include/solvers/CScalarSolver.inl
@@ -30,9 +30,6 @@
 #include "../../include/variables/CFlowVariable.hpp"
 
 template <class VariableType>
-CScalarSolver<VariableType>::CScalarSolver(bool conservative) : CSolver(), Conservative(conservative) {}
-
-template <class VariableType>
 CScalarSolver<VariableType>::CScalarSolver(CGeometry* geometry, CConfig* config, bool conservative)
     : CSolver(), Conservative(conservative) {
   nMarker = config->GetnMarker_All();

--- a/SU2_CFD/include/solvers/CTransLMSolver.hpp
+++ b/SU2_CFD/include/solvers/CTransLMSolver.hpp
@@ -39,12 +39,8 @@
 class CTransLMSolver final : public CTurbSolver {
 private:
   su2double Intermittency_Inf, REth_Inf;
-public:
-  /*!
-   * \brief Constructor of the class.
-   */
-  CTransLMSolver(void);
 
+public:
   /*!
    * \overload
    * \param[in] geometry - Geometrical definition of the problem.

--- a/SU2_CFD/include/solvers/CTurbSASolver.hpp
+++ b/SU2_CFD/include/solvers/CTurbSASolver.hpp
@@ -74,12 +74,7 @@ private:
 
 public:
   /*!
-   * \brief Constructor of the class.
-   */
-  CTurbSASolver();
-
-  /*!
-   * \overload
+   * \brief Constructor.
    * \param[in] geometry - Geometrical definition of the problem.
    * \param[in] config - Definition of the particular problem.
    * \param[in] iMesh - Index of the mesh in multigrid computations.

--- a/SU2_CFD/include/solvers/CTurbSSTSolver.hpp
+++ b/SU2_CFD/include/solvers/CTurbSSTSolver.hpp
@@ -37,10 +37,7 @@
  */
 class CTurbSSTSolver final : public CTurbSolver {
 private:
-  su2double
-  constants[10] = {0.0}; /*!< \brief Constants for the model. */
-
-
+  su2double constants[10] = {0.0}; /*!< \brief Constants for the model. */
 
   /*!
    * \brief Compute nu tilde from the wall functions.
@@ -56,15 +53,9 @@ private:
                      const CConfig *config,
                      unsigned short val_marker);
 
-
 public:
   /*!
-   * \brief Constructor of the class.
-   */
-  CTurbSSTSolver(void);
-
-  /*!
-   * \overload
+   * \brief Constructor.
    * \param[in] geometry - Geometrical definition of the problem.
    * \param[in] config - Definition of the particular problem.
    * \param[in] iMesh - Index of the mesh in multigrid computations.
@@ -310,5 +301,5 @@ public:
    */
   inline su2double GetOmega_Inf(void) const override { return Solution_Inf[1]; }
 
-  
+
 };

--- a/SU2_CFD/include/solvers/CTurbSolver.hpp
+++ b/SU2_CFD/include/solvers/CTurbSolver.hpp
@@ -49,12 +49,6 @@ protected:
 
 
 public:
-
-  /*!
-   * \brief Constructor of the class.
-   */
-  CTurbSolver(bool conservative);
-
   /*!
    * \brief Destructor of the class.
    */

--- a/SU2_CFD/src/solvers/CTransLMSolver.cpp
+++ b/SU2_CFD/src/solvers/CTransLMSolver.cpp
@@ -30,13 +30,13 @@
 #include "../../include/variables/CTurbSAVariable.hpp"
 
 /*---  This is the implementation of the Langtry-Menter transition model.
-       The main reference for this model is:Langtry, Menter, AIAA J. 47(12) 2009 
+       The main reference for this model is:Langtry, Menter, AIAA J. 47(12) 2009
        DOI: https://doi.org/10.2514/1.42362 ---*/
-       
-// Note: TransLM seems to use rho*gamma, rho*Re_sigma as Solution variables, thus Conservative=true
-CTransLMSolver::CTransLMSolver(void) : CTurbSolver(true) {}
 
-CTransLMSolver::CTransLMSolver(CGeometry *geometry, CConfig *config, unsigned short iMesh) : CTurbSolver(true) {
+// Note: TransLM seems to use rho*gamma, rho*Re_sigma as Solution variables, thus Conservative=true
+
+CTransLMSolver::CTransLMSolver(CGeometry *geometry, CConfig *config, unsigned short iMesh)
+    : CTurbSolver(geometry, config, true) {
   unsigned short iVar, nLineLets;
   unsigned long iPoint, index;
   su2double tu_Inf, dull_val, rey;

--- a/SU2_CFD/src/solvers/CTurbSASolver.cpp
+++ b/SU2_CFD/src/solvers/CTurbSASolver.cpp
@@ -32,8 +32,6 @@
 #include "../../../Common/include/toolboxes/geometry_toolbox.hpp"
 
 
-CTurbSASolver::CTurbSASolver(void) : CTurbSolver(false) { }
-
 CTurbSASolver::CTurbSASolver(CGeometry *geometry, CConfig *config, unsigned short iMesh, CFluidModel* FluidModel)
              : CTurbSolver(geometry, config, false) {
 

--- a/SU2_CFD/src/solvers/CTurbSSTSolver.cpp
+++ b/SU2_CFD/src/solvers/CTurbSSTSolver.cpp
@@ -32,8 +32,6 @@
 #include "../../../Common/include/toolboxes/geometry_toolbox.hpp"
 
 
-CTurbSSTSolver::CTurbSSTSolver(void) : CTurbSolver(true) { }
-
 CTurbSSTSolver::CTurbSSTSolver(CGeometry *geometry, CConfig *config, unsigned short iMesh)
     : CTurbSolver(geometry, config, true) {
   unsigned short nLineLets;

--- a/SU2_CFD/src/solvers/CTurbSolver.cpp
+++ b/SU2_CFD/src/solvers/CTurbSolver.cpp
@@ -33,16 +33,12 @@
 /*--- Explicit instantiation of the parent class of CTurbSolver. ---*/
 template class CScalarSolver<CTurbVariable>;
 
-CTurbSolver::CTurbSolver(bool conservative) : CScalarSolver<CTurbVariable>(conservative) { }
-
 CTurbSolver::CTurbSolver(CGeometry* geometry, CConfig *config, bool conservative) : CScalarSolver<CTurbVariable>(geometry, config, conservative) { }
 
 CTurbSolver::~CTurbSolver() {
-
   for (auto& mat : SlidingState) {
     for (auto ptr : mat) delete [] ptr;
   }
-
 }
 
 void CTurbSolver::BC_Riemann(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics, CNumerics *visc_numerics, CConfig *config, unsigned short val_marker) {

